server/src/admin/admin.router.ts

```diff
@@ -0,0 +1,268 @@
+    const result = await reportReview({ id });
+    if (!result) {
+      return res
+        .status(400)
+        .json({ error: `Review with id: ${id} unable to be reported.` });
+    }
+
+    return res
+      .status(200)
+      .json({ message: `Review with id: ${id} successfully reported.` });
+  } catch (err) {
+    return res
+      .status(500)
+      .json({ error: `Internal Server Error: ${err.message}` });
+  }
+});
+
+/*
+ * Check if a token is for an admin
+ */
+adminRouter.post('/tokenIsAdmin', async (req, res) => {
+  try {
+    const { token } = req.body;
+    const auth: Auth = new Auth({ token });
+
+    const isAdmin = await verifyTokenAdmin({ auth });
+
+    return res.status(200).json({ result: isAdmin });
+  } catch (err) {
+    return res.status(500).json({ error: `Internal Server Error: ${err}` });
+  }
+});
+
+adminRouter.post('/makeReviewVisible', async (req, res) => {
+  try {
+    const { token, review }: AdminReviewRequestType = req.body;
+    const auth = new Auth({ token });
+
+    if (review.reported !== 1) {
+      const reviewVisible = await editReviewVisibility({
+        reviewId: review._id,
+        auth,
+        visibility: 1,
+        reported: 0,
+      });
+
+      if (reviewVisible) {
+        return res.status(200).json({
+          message: `Review with id: ${review._id} is now visible!`,
+        });
+      }
+    } else {
+      return res.status(400).json({
+        error:
+          'Review has been reported, to make review visible please undo the report.',
+      });
+    }
+
+    return res.status(400).json({
+      error: `Review has invalid id: ${review._id} or user is not an admin.`,
+    });
+  } catch (err) {
+    return res.status(500).json({ error: `Internal Server Error: ${err}` });
+  }
+});
+
+adminRouter.post('/fetchPendingReviews', async (req, res) => {
+  try {
+    const { token }: AdminRequestType = req.body;
+    const auth = new Auth({ token });
+    const pendingReviews = await getPendingReviews({ auth });
+    if (pendingReviews === null) {
+      return res.status(400).json({
+        error: `User is not an admin.`,
+      });
+    }
+
+    return res.status(200).json({
+      message: 'Retrieved all pending reviews',
+      result: pendingReviews,
+    });
+  } catch (err) {
+    return res.status(500).json({ error: `Internal Server Error: ${err}` });
+  }
+});
+
+adminRouter.post('/getRaffleWinner', async (req, res) => {
+  try {
+    const { startDate }: RaffleWinnerRequestType = req.body;
+    const winner = await getRaffleWinner({ startDate });
+
+    if (winner === null) {
+      return res.status(400).json({
+        error: `No raffle winner found.`,
+      });
+    }
+
+    return res.status(200).json({
+      message: 'Retrieved raffle winner',
+      result: winner,
+    });
+  } catch (err) {
+    return res.status(500).json({ error: `Internal Server Error: ${err}` });
+  }
+});
+
+adminRouter.post('/addNewSemDbester', async (req, res) => {
```

Comment:
Believe this should be "/addNewSemester" - as frontend calls "/addNewSemester" unless I am mistaken. 

This is the one button on the frontend that would not work I believe.

Reply:
Thanks for catching that Will! Fixed!